app/models/mixins/authentication_mixin.rb

```diff
@@ -476,6 +472,7 @@ def authentication_check_no_validation(type, options)
 
   def authentication_best_fit(type = nil)
     # Look for the supplied type and if that is not found return the default credentials
+    type ||= default_authentication_type
```

Comment:
have to say, not a big fan of using defaults and special logic here.
If this were more simple/straight forward then we could use a regular association here.

but that is a future request

Reply:
I also wonder how often we actually pass in a type here.

e.g.: I notice that `auth_user_keypair` is only called with `type=nil`

again, this is fur future thought/pondering

Reply:
Yeah I don't disagree re: defaults but most of the time we don't pass a type in for the default case and this would bring back everything first then check the default which is what we are trying to avoid